api/src/main/java/jakarta/data/repository/BasicRepository.java

```diff
@@ -131,7 +132,7 @@ public interface BasicRepository<T, K> extends DataRepository<T, K> {
      * @throws NullPointerException If either the iterable is null or any element is null.
      */
     @Save
-    <S extends T> Iterable<S> saveAll(Iterable<S> entities);
+    <S extends T> List<S> saveAll(List<S> entities);
```

Comment:
This might be late, but could we keep the parameter as Iterable?
Thus, it can work with any input that we wish.

Reply:
Well, I mean, we _could_, but why? What's the concrete use case? 

The only use case I can see is that you want to pass a `Set` here. But `Set` just has the wrong semantics, because it's in general _not_ ordered, and so its elements can't be correlated with the elements in the resulting `List`.

And apart from `Set`, I just don't know any other type of thing that's `Iterable` that you might have a reasonable expectation of wanting to pass in here.

Reply:
And for another thing, I just want simple rules here: "use arrays or `List`s."

Not: "you can use `Iterable` here in this place but it has to be a `List` here in this other place".

Nor do I want to leave the user with this vague feeling of uncertainty about whether they should have used `List` or `Collection` or `Iterable`. In this case, I think it's better to just decide for them.

Reply:
Yes, it makes sense.
Thus, never mind.

Reply:
cool.